mods/default/nodes.lua

```diff
@@ -1761,6 +1762,218 @@ minetest.register_node("default:chest_locked", {
+		local inv = minetest.get_meta(pos):get_inventory()
+
+		if inv:is_empty("input") and inv:is_empty("nametag") and inv:is_empty("output") then
+			return true
+		else
+			return false
+		end
+	end,
+	on_blast = function(pos)
+		local inv = minetest.get_meta(pos):get_inventory()
+
+		local drops = {
+			inv:get_list("input")[1],
+			inv:get_list("nametag")[1],
+			inv:get_list("output")[1],
+			"default:workbench",
+		}
+		minetest.remove_node(pos)
+		return drops
+	end,
+
+	allow_metadata_inventory_put = function(pos, listname, index, stack)
+		local inv = minetest.get_meta(pos):get_inventory()
+
+		if listname == "nametag" then
+			if stack:get_name() ~= "default:nametag" then
+				return 0
+			else
+				return stack:get_count()
+			end
+		elseif listname == "output" then
+			return 0
+		elseif listname == "input" then
+			if stack:get_name() == "default:nametag" then
+				return 0
+			elseif not inv:is_empty("input") then
+				return 0
+			end
+		end
+
+		return 1
+	end,
+	allow_metadata_inventory_move = function(pos, from_list, from_index, to_list, to_index, count, player)
+		local inv = minetest.get_meta(pos):get_inventory()
+		local stack = inv:get_stack(from_list, from_index)
+
+		if to_list == "nametag" then
+			if stack:get_name() ~= "default:nametag" then
+				return 0
+			end
+		elseif to_list == "input" then
+			if stack:get_name() == "default:nametag" then
+				return 0
+			end
+		elseif to_list == "output" then
+			return 0
+		end
+
+		return count
+	end,
+	on_metadata_inventory_put = function(pos, listname, index, stack)
+		local meta = minetest.get_meta(pos)
+		local inv = meta:get_inventory()
+		local error = meta:get_string("error")
+
+		if error == "input" and not inv:is_empty("input") then
+			meta:set_string("formspec", get_workbench_formspec(pos))
+		elseif error == "nametag" and not inv:is_empty("nametag") then
+			meta:set_string("formspec", get_workbench_formspec(pos))
+		end
+
+		if listname == "input" then
+			workbench_update_text(pos, stack)
+		end
+	end,
+	on_metadata_inventory_move = function(pos, from_list, from_index, to_list, to_index)
+		local meta = minetest.get_meta(pos)
+		local inv = meta:get_inventory()
+		local stack = inv:get_stack(to_list, to_index)
+
+		if to_list == "input" then
+			workbench_update_text(pos, stack)
+		end
+	end,
+	on_metadata_inventory_take = function(pos, listname)
+		local meta = minetest.get_meta(pos)
+		local inv = meta:get_inventory()
+		local error = meta:get_string("error")
+
+		if error == "output" and inv:is_empty("output") then
+			meta:set_string("formspec", get_workbench_formspec(pos))
+		end
+
+		if listname == "input" then
+			meta:set_string("text", "")
+			meta:set_string("formspec", get_workbench_formspec(pos))
+		end
+	end,
+	on_receive_fields = function(pos, formname, fields, sender)
+		local meta = minetest.get_meta(pos)
+		local inv = meta:get_inventory()
+
+		if fields.rename or fields.key_enter_field == "text" then
+			meta:set_string("text", fields.text)
```

Comment:
This should be limited to perhaps 1000 characters. I believe these nametags aren't supposed to be whole books.